### PR TITLE
146 v template

### PR DESCRIPTION
**Pull request by smgl9**

### CODE REVIEW

File: src/templates/verilogeditor.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ class Verilog_editor {
     str += this.set_Constants(space, structure['generics']);
     str += '\n';
     str += '  // Ports\n';
-    str += this.set_Signals(space, structure['ports']);
+    str += this.set_Signals_Tb(space, structure['ports']);
```
Comment (qarlosalberto) [inline src/templates/verilogeditor.js:58]: Try to use lower case.

File: src/templates/verilogeditor.js [hunk 2]
```diff
@@ -201,10 +236,16 @@ class Verilog_editor {
     return str;
   }
 
-  set_Clk_Process(space) {
+  set_Clk_Process(space, ports) {
     var str = '';
-    str += '// ' + space + "always\n";
-    str += '// ' + space + "  #5  clk =  ! clk;\n";
+    for (let x = 0; x < ports.length - 1; ++x) {
```
Comment (qarlosalberto) [inline src/templates/verilogeditor.js:241]: Why -1 in **x < ports.length - 1**?

File: src/templates/verilogeditor.js [hunk 3]
```diff
@@ -111,11 +111,46 @@ class Verilog_editor {
...
+        const regex = /\[(.*?)\]/;
+        let type = m[x]['type'].match(regex);
+        if (type === null) {
+          type = '';
+        } else {
+          type = type[0];
+        }
+        if (m[x]['direction'] === "input") {
+          if (type === '') {
+            str += space + 'reg ' + type + ' ' + m[x]['name'] + ' = 0;\n';
```
Comment (qarlosalberto) [inline src/templates/verilogeditor.js:135]: Try to use "`" and " `". Example: 

str += \`${space} reg ${type} ${m[x]['name']} = 0;\n;\`

File: src/templates/verilogeditor.js (hunk 1 above)
Comment (smgl9) [reply src/templates/verilogeditor.js:58]: I have no problems using lower case. But as you know we've changed the style. So it means changing the whole file.
No problem at all but just to let you know.

Comment (qarlosalberto) [reply src/templates/verilogeditor.js:58]: Yes, but in the new code added we should use the new style.

File: src/templates/verilogeditor.js (hunk 2 above)
Comment (smgl9) [reply src/templates/verilogeditor.js:241]: No reason, my mistake

File: src/templates/verilogeditor.js (hunk 3 above)
Comment (smgl9) [reply src/templates/verilogeditor.js:135]: Same comment than before. I consider the style refactoring a separate issue

Comment (qarlosalberto) [reply src/templates/verilogeditor.js:135]: That is ok with the old code, but no with the new code added. Anyway, I think that isn't an old or new style problem. In this case In this case I think the code is cleaner with \`

File: src/templates/verilogeditor.js (hunk 1 above)
Comment (smgl9) [reply src/templates/verilogeditor.js:58]: I agree, but this is old code and bug fixing. Not a refactoring.

**Review by qarlosalberto — APPROVED** [review]

**Pull request merged**